docs/markdown/snippets/msvc_argument_changes.md

```diff
@@ -0,0 +1,20 @@
+## MSVC/Clang-Cl Argument Changes/Cleanup
+
+* "Edit and continue" (`/ZI`) is no longer used by default for Visual Studio debug. Replaced by /Zi.
+* "Disable Debug" (`/Od`) is no longer manually specified for optimization levels {`0`,`g`} (it is already the default for MSVC).
+* "Run Time Checking" (`/RTC1`) removed from `debug` buildtype by default
+* Clang-CL `debug` buildtype arguments now match MSVC arguments
+* There is now no difference between `buildtype` flags and `debug` + `optimization` flags
+
+Meson was adding the `/ZI` compiler argument as an argument for Visual Studio in debug mode. This enables the `edit-and-continue` debugging in Visual Studio IDE's.
+Unfortunately, it is also extremely expensive and breaks certain use cases such as link time code generation. Edit and continue can be enabled by manually by adding `/ZI` to compiler arguments. The `/ZI` argument has now been replaced by the `/Zi` argument for debug builds.
+
+The /Od flag has been removed, as it is already the default in the MSVC compilers, and conflicts with other user options.
+
+/RTC1 conflicts with other RTC argument types as there are many different options, and has been removed by default.
+Run Time Checking can be enabled by manually adding `/RTC1` or other RTC flags of your choice.
```

Comment:
Changing the scope of a MR and adding new functionality after review is not really nice. Changes like these should be done as separate MRs or, at the very least, be noticeably mentioned in the discussion page. Please restore `/RTC1` and let's do that in a separate MR properly. There seem to be a way to define Windows-specific things like RTC level in one place so that it is used with both MSVC and Clang both with C and c++. That should not block merging this one, though, as the`/ZI` change is clearly a separate issue.

Reply:
My apologies, I had originally thought about making them separate, but I thought they were related enough to not justify it.
I reverted and will make a new PR with the argument cleanup.